src/main/java/co/rsk/federate/btcreleaseclient/BtcReleaseClient.java

```diff
@@ -185,42 +200,57 @@ public void stop(Federation federation) {
 
     @PreDestroy
     public void tearDown() {
-        org.bitcoinj.core.Context.propagate(new org.bitcoinj.core.Context(ThinConverter.toOriginalInstance(bridgeConstants.getBtcParamsString())));
+        org.bitcoinj.core.Context.propagate(
+            new org.bitcoinj.core.Context(
+                ThinConverter.toOriginalInstance(bridgeConstants.getBtcParamsString())));
         peerGroup.stop();
         peerGroup = null;
     }
 
     private class BtcReleaseEthereumListener extends EthereumListenerAdapter {
         @Override
         public void onBestBlock(org.ethereum.core.Block block, List<TransactionReceipt> receipts) {
+            if (!isPegoutEnabled) {
+                return;
+            }
```

Comment:
you sure about this?

Reply:
Yes, if no pegouts are enabled there should be no point in signing.

Reply:
I think we should add a comment explaining the purpose of this field in the configuration class and also the conf file sample.

Reply:
Good idea, lets create a task for this.